todo-app/src/App.js

```diff
@@ -1,13 +1,109 @@
 import React, { Component } from 'react';
+import Task from './Task';
+//import tasks from './task.json';
+import TaskForm from './taskForm';
+import Moment from 'moment';
 
-class App extends Component {
+export default class App extends Component {
+
+  constructor(props) {
+    super(props);
+	let tasks = [];
+	if (localStorage.tasks != null) { 
+		tasks = JSON.parse(localStorage.tasks);
+	}
+    this.state = {
+      tasksArr: tasks,
+	  editingTaskID: null
+    };
+
+    this.handleDoneChange = this.handleDoneChange.bind(this);
+  }
+  
+  handleDoneChange(doneValue, taskId) {
+    const updatedtasks = this.state.tasksArr.map(task => {
+      if (taskId === task.id) {
+        return Object.assign({}, task, {done: doneValue});
+      }
+      return task;
+    });
+    this.setState({tasksArr: updatedtasks});
+	localStorage.tasks = JSON.stringify(updatedtasks);
+  }
+  
+  handleTaskAdd = taskText => {
+	const ids = this.state.tasksArr.map(task => task.id);
+    const newTask =  {
+      id:     ids.length === 0 ? 1 : Math.max(...ids) + 1,
+      description: taskText,
+      deadline:   Moment().format('YYYY-MM-DD'),
```

Comment:
What happened here? Each task is supposed to have a description and a due date/deadline. Your `deadline` property here is more of a `createdAt` timestamp - there's no way for us to set the deadline to any date other than today.